components/toggle/Toggle.js

```diff
@@ -0,0 +1,64 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import theme from './theme.css';
+import cx from 'classnames';
+import omit from 'lodash.omit';
+
+class Toggle extends PureComponent {
+  static propTypes = {
+    checked: PropTypes.bool,
+    disabled: PropTypes.bool,
+    name: PropTypes.string,
+    className: PropTypes.string,
+    onChange: PropTypes.func,
+    size: PropTypes.oneOf(['small', 'medium', 'large']),
+    color: PropTypes.oneOf(['neutral', 'teal', 'ruby', 'mint', 'violet', 'gold', 'aqua']),
+  };
+
+  static defaultProps = {
+    checked: false,
+    disabled: false,
+    size: 'medium',
+    color: 'neutral',
+  };
+
+  constructor() {
+    super(...arguments);
+    this.handleToggle = ::this.handleToggle;
+  }
+
+  handleToggle() {
+    if (!this.props.disabled && this.props.onChange) {
+      this.props.onChange(!this.props.checked, event);
+    }
+  }
+
+  render() {
+    const { checked, disabled, className, size, color, ...others } = this.props;
+    const rest = omit(others, ['onChange']);
+
+    const classNames = cx(theme['toggle'], className, theme[size], theme[color], {
+      [theme['checked']]: checked,
+      [theme['disabled']]: disabled,
+    });
+
+    return (
+      <label data-teamleader-ui="toggle" className={classNames}>
```

Comment:
The component is wrapped in a **label** tag, but I don't see any label prop being rendered?

Reply:
Well, it's a label to make "invisible checkbox" work. It's just a small trick.

Reply:
Ok, good stuff! But I think we'll need a label (rendered next to the toggle) as well?

![schermafdruk 2017-10-19 09 35 28](https://user-images.githubusercontent.com/5336831/31759385-81af852c-b4b1-11e7-8b39-18b666d11c99.png)

Reply:
yes, I just still need to add it

Reply:
:ok: by @lowiebenoot on 26/10/2017 10:22:53